cmd/allocator/main.go

```diff
@@ -106,6 +107,30 @@ func main() {
 			logger.WithError(err).Fatalf("cannot watch folder %s for secret changes", certDir)
 		}
 
+		go func() {
+			for {
+				select {
+				// watch for events
+				case event := <-watcher.Events:
+					h.certMutex.Lock()
+					caCertPool, err := getCACertPool(certDir)
+					if err != nil {
+						logger.WithError(err).Error("could not load CA certs; keeping old ones")
+					} else {
+						h.caCertPool = caCertPool
+					}
+					logger.Infof("Certificate directory change event %v", event)
+					h.certMutex.Unlock()
+
+				// watch for errors
+				case err := <-watcher.Errors:
+					logger.WithError(err).Error("error watching for certificate directory")
+				}
+			}
+		}()
+	}
+
+	if !h.tlsDisabled {
```

Comment:
tlsDisabled is the superset.

```
if !h.tlsDisabled {
  watch for tlsCert
  if !h.mTLSDisabled {
    watch for caCertPool
  }
}
```

Reply:
Sure, that makes sense. I thought we would want two independent flags, but you can't really have mtls without at least tls. Done!